apps/console/src/features/roles/components/wizard-updated/application-role-wizard.tsx

```diff
@@ -386,49 +392,69 @@ export const ApplicationRoleWizard: FunctionComponent<ApplicationRoleWizardProps
+                        groupBy={ (apiResourcesListOption: DropdownItemProps) =>
+                            APIResourceUtils
+                                .resolveApiResourceGroup(apiResourcesListOption?.type) }
+                        isOptionEqualToValue={
+                            (option: DropdownItemProps, value: DropdownItemProps) =>
+                                option.value === value.value
                         }
-                        loading={ false }
-                        onChange={ onAPIResourceSelected }
-                        hint={
-                            !isSubscribedAPIResourcesListLoading
-                            && apiResourcesListOptions?.length === 0
-                            && selectedAPIResources?.length === 0
-                            && (
-                                <Hint>
-                                    <Trans
-                                        i18nKey={ "console:manage.features.roles.addRoleWizard.forms.rolePermission" +
-                                            ".apiResource.hint.empty" }
-                                    >
-                                        There are no API resources authorized for the selected application.
-                                        API Resources can be authorized through <Link
-                                            external={ false }
-                                            onClick={ () => {
-                                                history.push(
-                                                    `${ AppConstants.getPaths()
-                                                        .get("APPLICATION_EDIT").replace(":id", appId) }#tab=4`
-                                                );
-                                            } }
-                                        > here </Link>
-                                    </Trans>
-                                </Hint>
+                        options={
+                            apiResourcesListOptions?.filter((item: DropdownProps) =>
+                                item?.type === APIResourceCategories.TENANT ||
+                                item?.type === APIResourceCategories.ORGANIZATION ||
+                                item?.type === APIResourceCategories.BUSINESS
+                            ).sort((a: DropdownProps, b: DropdownProps) =>
+                                -b?.type?.localeCompare(a?.type)
                             )
                         }
+                        onChange={ onAPIResourceSelected }
+                        noOptionsText={ isSubscribedAPIResourcesListLoading
+                            ? t("common:searching")
+                            :  t("common:noResultsFound")
+                        }
+                        renderInput={ (params: AutocompleteRenderInputParams) => (
+                            <TextField
+                                { ...params }
```

Comment:
Let's move this to the bottom of the prop list

Reply:
Moved